### ISSUE TITLE
Cannot interpolate tag key name

### ISSUE THREAD

**Issue by sethvargo** [hashicorp/terraform#11324]
### Terraform Version
v0.8.5-dev

### Affected Resource(s)
- tags
- probably core, since it's interpolation

### Terraform Configuration Files
```hcl
resource "random_id" "tag_key" {
  byte_length = "16"
}

resource "random_id" "tag_value" {
  byte_length = "32"
}

resource "aws_instance" "server" {
  count = "${var.servers}"

  ami           = "${data.aws_ami.ubuntu-1404.id}"
  instance_type = "r3.large"
  key_name      = "${aws_key_pair.lab.id}"

  subnet_id              = "${element(aws_subnet.lab.*.id, count.index)}"
  vpc_security_group_ids = ["${aws_security_group.lab.id}"]

  tags {
    "${random_id.tag_key.hex}" = "${random_id.tag_value.hex}"
  }
}
```

### Expected Behavior
Instance should be created with tags from the values of random_id.

### Actual Behavior
The tag is created as (literally) `${random_id.tag_key.hex}`, and the value is empty.

### Steps to Reproduce
1. `terraform apply`

**Comment by sethvargo** [comment]
<img width="432" alt="screen shot 2017-01-20 at 6 04 10 pm" src="https://cloud.githubusercontent.com/assets/408570/22168474/dfdb246e-df3a-11e6-9e12-8f87858ae545.png">

**Comment by apparentlymart** [comment]
Hey @sethvargo!

Agreed that this ought to work the way you wrote it, but I think you should be able to work around it by using some more awkward syntax:

```
tags = "${map(random_id.tag_key.hex, random_id.tag_value.hex)}"
```

I didn't actually test this so maybe there's something else that'll break there but I *think* that should work.

**Comment by mvisonneau** [comment]
Related to https://github.com/hashicorp/terraform/issues/14516

**Comment by sethvargo** [comment]
This is fixed in Terraform 0.12 (beta) slash HCL2
